compass/ocean/tests/nonhydro/solitary_wave/visualize.py

```diff
@@ -0,0 +1,63 @@
+import xarray
+import numpy as np
+from netCDF4 import Dataset
+from scipy.interpolate import griddata
+from compass.step import Step
+import matplotlib.pyplot as plt
+import matplotlib
+from matplotlib import transforms
+matplotlib.use('Agg')
+
+
+class Visualize(Step):
+    """
+    A step for visualizing a cross-section through the solitary wave
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+        """
+        super().__init__(test_case=test_case, name='visualize')
+
+        for grid in ['nonhydro', 'hydro']:
+            self.add_input_file(filename=f'output_{grid}.nc',
+                                target=f'../{grid}/output.nc')
+            self.add_input_file(filename=f'init_{grid}.nc',
+                               target=f'../{grid}/init.nc')
+        self.add_output_file('plotTemp.png')
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        grids = ['nonhydro', 'hydro']
+        nGrids = len(grids)
+        plt.figure(1, figsize=(12.0, 6.0))
+
+        for j in range(nGrids):
+            grid = grids[j] 
+            ncfileIC = Dataset(f'init_{grid}.nc', 'r')
+            ncfile = Dataset(f'output_{grid}.nc', 'r')
+            temp = ncfile.variables['temperature'][27, 0:1200, :]
+            xCell = ncfileIC.variables['xCell'][0:1200]
+            zMid = ncfile.variables['zMid'][27, 0, :]
+            L0 = 1436
+            a0 = 220
+            x = xCell/L0
+            z = zMid/a0
+            z1 = z[0:35]
+            temp1 = temp[:, 0:35]
```

Comment:
Probably the same here, 35 should come from a config option.

Reply:
Done.